app/models/update_log.rb

```diff
@@ -1,17 +1,60 @@
+# frozen_string_literal: true
+
 class UpdateLog
+  ####################
+  # CONSTANTS        #
+  ####################
+  # We want to keep a maximum number of 10 updates, so the last key we want to save is the ninth
+  MAX_UPDATES_KEY = 9
 
-  def self.setting_record
-    Setting.find_or_create_by(key: 'metrics_update')
+  def self.log_updates(starttime, endtime)
```

Comment:
I think this interface is too brittle. If there is more we want to save, we'll continually need to change the expected arguments along with the logic for saving it.

I suggest instead that we take in all the data we want to save as a hash, and then just save it like:

```ruby
setting_record.value['constant_update'][this_update] = input_hash
```

That hash will have keys for the current things we require (start_time, end_time) but it will then be easy to save more data without breaking anything.